src/snowflake/connector/result_batch.py

```diff
@@ -65,7 +62,9 @@ def _create_vendored_arrow_iterator(
     numpy: bool,
     number_to_decimal: bool,
     row_unit: IterUnit,
-) -> PyArrowIterator:
+):
+    from .arrow_iterator import PyArrowIterator
```

Comment:
Will this raise an error if PyArrow isn't compiled?

Reply:
no, the `_create_vendored_arrow_iterator` and `_create_nanoarrow_iterator` will only be called when we detect that arrow can be used.